crates/fuel-gas-price-algorithm/src/v1.rs

```diff
@@ -153,11 +161,11 @@ pub struct AlgorithmUpdaterV1 {
     /// The maximum percentage that the DA portion of the gas price can change in a single block
     pub max_da_gas_price_change_percent: u8,
```

Comment:
nit: I feel like introducing a newtype to enforce the percentage invariant (in range `0..=100`) might be a good idea. Up to you.

Reply:
I'm open to it! I'd just have to introduce fallibility into the constructor.

Also. In theory, this could be set to something larger than 100%. So even `u8` is an arbitrary ceiling to how high it could be set to. I'm realizing that's why `exec_gas_price_change_percent` wasn't `u8` before. Both could be higher in theory.

The `l2_block_fullness_threshold_percent` on the otherhand would be a better candidate for a newtype since there is no such thing as a 110% full block hopefully.

Reply:
The arbitrary u8 ceiling feels weird, if there's ever any reason to set it to 200%, surely there's also reasons to set it to 300% as well? In any case, it wasn't clear to me from the doc comment, so maybe explicitly writing it there would be a good first step, and the newtype can be introduced later when the implementation has settled down.

Reply:
Changed both to `u16` and added explanation.

Reply:
Added a newtype for the threshold percentage. I called it `ClampedPercent` to communicate that it's infallible and just clamps the value.